packages/core/src/routes/sign-in-experience.test.ts

```diff
@@ -113,6 +115,24 @@ describe('PATCH /sign-in-exp', () => {
     });
   });
 
+  it('should filter out unavailable social connector targets', async () => {
+    const signInMethods = { ...mockSignInMethods, social: SignInMethodState.Secondary };
+    const socialSignInConnectorTargets = ['github', 'facebook', 'google'];
+    const signInExperience = {
+      signInMethods,
+      socialSignInConnectorTargets,
+    };
+    const response = await signInExperienceRequester.patch('/sign-in-exp').send(signInExperience);
+    expect(response).toMatchObject({
+      status: 200,
+      body: {
+        ...mockSignInExperience,
+        signInMethods,
+        socialSignInConnectorTargets: ['github', 'facebook'],
+      },
+    });
```

Comment:
better use `expect.objectContaining({…})` just match the specific part
```suggestion
    expect(response).toMatchObject(
      expect.objectContaining({
        status: 200,
        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
        body: expect.objectContaining({
          socialSignInConnectorTargets: ['github', 'facebook'],
        }),
      })
    );
```

Reply:
Let's keep the old logic.

Reply:
nit: I'm good with both. Seems that `expect.objectContaining({…})` will be easier to maintain the test cases.